WooCommerce/Classes/POS/ViewModels/TotalsViewModel.swift

```diff
@@ -86,7 +87,38 @@ final class TotalsViewModel: ObservableObject {
         startSyncingOrder(with: cartItems, allItems: allItems)
     }
 
+    private func areOrderAndCartDifferent(cartItems: [CartItem]) -> Bool {
+        // check if order has same items as cart does
+        if let order {
+            // first we get list of all products 1 by 1
+            var cleanOrderItems: [POSOrderItem] = order.items.flatMap {
+                Array(repeating: $0, count: $0.quantity.intValue)
+            }
+            var cleanCartItems: [POSItem] = cartItems.flatMap {
+                Array(repeating: $0.item, count: $0.quantity)
+            }
+            if cleanOrderItems.count == cleanCartItems.count {
+                // sort items by productIDs to have them in same order for comparison
+                cleanOrderItems.sort { $0.productID < $1.productID }
+                cleanCartItems.sort { $0.productID < $1.productID }
+                // check if all the items are same, prices included
+                for (index, cartItem) in cleanCartItems.enumerated() {
+                    let orderItem = cleanOrderItems[index]
+                    if cartItem.productID != orderItem.productID || cartItem.price != orderItem.price.stringValue {
+                        return true
+                    }
```

Comment:
Question: Would be enough with the `productID` here? If the product has the same ID it shouldn't have different pricing for different instances. Also related to the previous comment regarding `cartItem` not having a `price` getter, if this is expected and we still need the price check then we should rename cartItem to item.

Reply:
Gotcha, that's definitely a good point. In this case I think would be good to extract this logic into a check which forces us to pass both the `productID` and the `price` with a comment of why is that needed, this way we're sure to not remove the price check in the future, or introduce regressions.